common/environment/environment.cpp

```diff
@@ -1732,6 +1822,51 @@ unsigned CConstDropZoneIteratorByComputer::count() const
 
 //--------------------------------------------------
 
+
+CConstDropZoneServerInfoIterator::CConstDropZoneServerInfoIterator(const IConstDropZoneInfo * dropZone)
+{
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
+    constEnv.setown((CLocalEnvironment *)factory->openEnvironment());
+    serverListIt.setown(dropZone->getPTree().getElements("ServerList"));
+    maxIndex = dropZone->getPTree().getCount("ServerList");
+}
+
+bool CConstDropZoneServerInfoIterator::first()
+{
+    bool hasFirst = serverListIt->first();
+    if (hasFirst)
+        curr = new CConstDropZoneServerInfo(constEnv, &serverListIt->query());
+    return hasFirst;
+}
+
+bool CConstDropZoneServerInfoIterator::next()
+{
+    bool hasNext = serverListIt->next();
+    if (hasNext)
+        curr = new CConstDropZoneServerInfo(constEnv, &serverListIt->query());
+    return hasNext;
+}
+
+bool CConstDropZoneServerInfoIterator::isValid()
+{
```

Comment:
is curr is set (by first() or next()) then it's valid, so would be more correct to do return nullptr != curr, instead of code below.